scaletest/notifications/config.go

```diff
@@ -0,0 +1,76 @@
+package notifications
+
+import (
+	"sync"
+	"time"
+
+	"golang.org/x/xerrors"
+
+	"github.com/google/uuid"
+
+	"github.com/coder/coder/v2/scaletest/createusers"
+)
+
+type Config struct {
+	// User is the configuration for the user to create.
+	User createusers.Config `json:"user"`
+
+	// IsOwner indicates if this user should be assigned Owner role.
+	// If true, the user will receive notifications.
+	IsOwner bool `json:"is_owner"`
+
+	// NotificationTimeout is how long to wait for notifications after triggering.
+	NotificationTimeout time.Duration `json:"notification_timeout"`
+
+	// DialTimeout is how long to wait for websocket connection.
+	DialTimeout time.Duration `json:"dial_timeout"`
+
+	// ExpectedNotifications maps notification template IDs to channels
+	// that receive the trigger time for each notification.
+	ExpectedNotifications map[uuid.UUID]chan time.Time `json:"-"`
+
+	Metrics *Metrics `json:"-"`
+
+	// DialBarrier ensures all runners are connected before notifications are triggered.
+	DialBarrier *sync.WaitGroup `json:"-"`
+
+	// OwnerDialBarrier is the barrier for owner users. Regular users wait on this to disconnect after owner users complete.
+	OwnerDialBarrier *sync.WaitGroup `json:"-"`
```

Comment:
```suggestion
	OwnerWatchBarrier *sync.WaitGroup `json:"-"`
```
because owners aren't dialing while we wait for this